src/parser/demonhunter/havoc/modules/spells/BladeDance.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SPECS from 'game/SPECS';
+import Events, { DamageEvent, FightEndEvent } from 'parser/core/Events';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+import SpellLink from 'common/SpellLink';
+
+//For Blade dance and Death Sweep
+class BladeDance extends Analyzer {
+  get suggestionThresholds() {
+    return {
+      actual: this.badCast,
+      isGreaterThan: {
+        minor: 0,
+        average: 0,
+        major: 1,
+      },
+      style: ThresholdStyle.NUMBER,
+    };
+  }
+
+  badCast = 0;
+  hitCount = 0;
+  hitTimeStamp: number = 0;
```

Comment:
The only place you use this is in a listener meaning it doesn't need to be global.